boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -139,32 +129,38 @@ mb1:
     jnz mb1
 
 mb2:
+#ifndef _M_IX86
+/* If the compiler is 64 bits then skip the 32 bits prefix using near jump we are already in protected mode */
+    .byte HEX(E9)
+    .long 1
+#endif
```

Comment:
The 32 bit prefix has nothing to do with being in protected mode, but comes from overwriting the default of 64 bit addresses in 64 bit mode (ebx instead of rbx). But since all code lives in the low address range, ebx should be ok in 64 bit mode, too. It should work without bypassing that prefix. If it really is needed, please use something less hacky, like
```
#ifndef _M_AMD64
    test dword ptr [rbx + MB_INFO_FLAGS_OFFSET], MB_INFO_FLAG_BOOT_DEVICE
#else
    test dword ptr [ebx + MB_INFO_FLAGS_OFFSET], MB_INFO_FLAG_BOOT_DEVICE
#endif
```